cylc/flow/task_events_mgr.py

```diff
@@ -526,8 +526,7 @@ def process_message(
                 itask, ("message %s" % str(severity).lower()), message)
         lseverity = str(severity).lower()
         if lseverity in self.NON_UNIQUE_EVENTS:
-            itask.non_unique_events.setdefault(lseverity, 0)
-            itask.non_unique_events[lseverity] += 1
+            itask.non_unique_events.update(lseverity)
```

Comment:
```suggestion
            itask.non_unique_events.update({lseverity: 1})
```
Fixes the broken test.

Reply:
Er, heck, thats confusing, the two should do the same:

```python
>>> from collections import Counter
>>> c = Counter()

>>> c.update({'a': 1})
>>> c
Counter({'a': 1})

>>> c.update('b')
>>> c
Counter({'a': 1, 'b': 1})
```

Reply:
Try it with a multi-character string.